test/unit_tests/braket/experimental/autoqasm/test_parameters.py

```diff
@@ -435,58 +486,61 @@ def test_bind_empty_program():
     def empty_program():
         pass
 
-    qasm = empty_program().to_ir()
-    bound_program1 = empty_program().make_bound_program({}).to_ir()
-    bound_program2 = empty_program().make_bound_program({"alpha": 0.5}).to_ir()
+    qasm = empty_program.to_ir()
+    bound_program1 = empty_program.make_bound_program({}).to_ir()
+    bound_program2 = empty_program.make_bound_program({"alpha": 0.5}).to_ir()
     assert qasm == bound_program1 == bound_program2
 
 
 def test_strict_parameter_bind():
     """Test make_bound_program with strict set to True."""
 
     @aq.main
-    def parametric(theta: float):
-        rx(0, theta)
+    def parametric(alpha: float):
+        rx(0, alpha)
         measure(0)
 
-    prog = parametric(FreeParameter("alpha"))
-
     template = """OPENQASM 3.0;
 float[64] alpha = {};
 qubit[1] __qubits__;
 rx(alpha) __qubits__[0];
 bit __bit_0__;
 __bit_0__ = measure __qubits__[0];"""
 
-    bound_prog = prog.make_bound_program({"alpha": 0.5}, strict=True)
+    bound_prog = parametric.make_bound_program({"alpha": 0.5}, strict=True)
     assert bound_prog.to_ir() == template.format(0.5)
 
 
 def test_strict_parameter_bind_failure():
     """Test make_bound_program with strict set to True."""
 
     @aq.main
-    def parametric(theta: float):
-        rx(0, theta)
+    def parametric(alpha: float):
+        rx(0, alpha)
         measure(0)
 
-    prog = parametric(FreeParameter("alpha"))
     with pytest.raises(
         aq.errors.ParameterNotFoundError, match="No parameter in the program named: beta"
     ):
-        prog.make_bound_program({"beta": 0.5}, strict=True)
+        parametric.make_bound_program({"beta": 0.5}, strict=True)
 
 
 def test_duplicate_variable_name_fails():
     """Test using a variable and FreeParameter with the same name."""
 
-    @aq.main
-    def parametric():
-        alpha = aq.FloatVar(1.2)  # noqa: F841
-        rx(0, FreeParameter("alpha"))
+    with pytest.raises(RuntimeError, match="conflicting variables with name alpha"):
+
+        @aq.main
+        def parametric_explicit():
+            alpha = aq.FloatVar(1.2)  # noqa: F841
+            rx(0, FreeParameter("alpha"))
 
     with pytest.raises(RuntimeError, match="conflicting variables with name alpha"):
-        parametric()
+
+        @aq.main
+        def parametric(alpha):
+            alpha = aq.FloatVar(1.2)  # noqa: F841
```

Comment:
Nice solution for covering existing tests and also focusing on the expected user interface. I appreciate it!